rs/nns/governance/src/governance.rs

```diff
@@ -4522,20 +4521,60 @@ impl Governance {
         );
     }
 
-    fn perform_manage_network_economics(&mut self, new_network_economics: NetworkEconomics) {
-        let Some(original_network_economics) = &self.heap_data.economics else {
-            // This wouldn't happen in production, but if it does, we try to do
-            // the best we can, because doing nothing seems more catastrophic.
+    fn perform_manage_network_economics(
+        &mut self,
+        proposal_id: u64,
+        proposed_network_economics: NetworkEconomics,
+    ) {
+        // This does all the "real work". This may look a little strange, but it
+        // ensures that we do NOT forget to call
+        // self.set_proposal_execution_status (e.g. when we return early). let
+        let mut main = || -> Result<(), GovernanceError> {
```

Comment:
I'd move this into a separate function, e.g., `perform_manage_network_economics_impl`

I think having a function that's 99% another function, +1 statement is more atypical